data-plane/benchmarks/run.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+
+# Copyright © 2018 Knative Authors (knative-dev@googlegroups.com)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+TIME=$(date +%s)
+
+run_java_filter_benchmarks_for_class() {
+  CLASS=$1
+
+  echo "Running benchmarks for ${CLASS}"
+
+  java -Dlogback.configurationFile=${SCRIPT_DIR}/resources/config-logging.xml \
+    -jar "${SCRIPT_DIR}/target/benchmarks.jar" $CLASS 2>&1 | tee "${SCRIPT_DIR}/output/${CLASS}.${TIME}.out.txt"
+
+  echo "Successfully ran benchmarks for ${CLASS}!\n\nThe results can be found at ${SCRIPT_DIR}/output/${CLASS}.${TIME}.out.txt"
+}
+
+SCRIPT_DIR=$(cd -- "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd)
+
+DATA_PLANE_DIR="${SCRIPT_DIR}/../"
+
+# check if the benchmark class exists
+
+if [[ ! -z $1 ]]; then
+  FOUND=0
+  while IFS="" read -r p || [[ -n "$p" ]]; do
+    if [[ "$1" == "$p" ]]; then
+      FOUND=1
+      break
+    fi
+  done <"${SCRIPT_DIR}/resources/filter-class-list.txt"
```

Comment:
Not an issue, but more a question: why didn't you use something like `egrep -q "^${1}$"`

Reply:
Just didn't think of it as I had already written the loop to go through the lines of `filter-class-list.txt` to run the benchmarks for all of the classes :smile:

Reply:
@creydr do you think I should switch? Your solution does seem cleaner to me

Reply:
Whatever you prefer. I am fine with both and could change it later anyhow

Reply:
I think i'll leave it for now since it keeps the script more consistent with itself